pkg/collector/pod_collector.go

```diff
@@ -89,12 +91,27 @@ func (c *PodCollector) GetMetrics() ([]CollectedMetric, error) {
 
 	ch := make(chan CollectedMetric)
 	errCh := make(chan error)
+	skippedPodsCount := 0
+
 	for _, pod := range pods.Items {
-		go c.getPodMetric(pod, ch, errCh)
+		t := time.Now()
+		podAge := time.Duration(t.Sub(pod.ObjectMeta.CreationTimestamp.Time).Nanoseconds())
+
+		if podAge > c.minPodAge {
+			if IsPodReady(pod) {
```

Comment:
Considering that we filter for PodReady which makes sense, I think the `min-pod-age` should actually be `min-pod-ready-age` meaning the minimum time since the pod got ready as opposed to the time since the pod was created. If you only consider the pod creation time you can in theory have a pod pending for hours and then when it gets ready at some point the `min-pod-age` will be satisfied right away even though the container may just have started.

The pod conditions have a `lastTransitionTime` field which can be used to determine how long a pod was ready.

Reply:
Yes, this makes sense. I'll try to figure out how to calculate that time.

Reply:
Hello, now it takes lastTransitionTime as a target for min-pod-ready-age calculation